tools/make/tests.toml

```diff
@@ -31,3 +31,47 @@ description = "Run all Rust doctests with all features"
 category = "ICU4X Development"
 command = "cargo"
 args = ["test", "--all-features", "--doc", "--no-fail-fast"]
+
+[tasks.test-cargo-tutorial-local]
+description = "Build the Cargo tutorial projects based on local crates"
+category = "ICU4X Development"
+script_runner = "@duckscript"
+script = '''
+exit_on_error true
+
+cd docs/tutorials/cargo_tests
+rm -f Cargo.lock
+
+pwd = pwd
+set_env CARGO_TARGET_DIR ${pwd}/target
+
+exec --fail-on-error cargo --config patch.toml run -p demo_baked
+exec --fail-on-error make -C buffer
+exec --fail-on-error cargo --config patch.toml run -p demo_buffer
+exec --fail-on-error cargo --config patch.toml run -p demo_experimental
+exec --fail-on-error cargo --config patch.toml run -p demo_lazy_static
+exec --fail-on-error cargo --config patch.toml run -p demo_testdata
+'''
+
+[tasks.test-cargo-tutorial-cratesio]
+description = "Build the Cargo tutorial projects based on crates.io"
+category = "ICU4X Development"
+script_runner = "@duckscript"
+script = '''
+exit_on_error true
+
+cd docs/tutorials/cargo_tests
+rm -f Cargo.lock
+
+pwd = pwd
+set_env CARGO_TARGET_DIR ${pwd}/target
+
+# Note: This uses the --release profile to distinguish build artifacts from the local job
```

Comment:
mmh do we plan on running this in CI?

Reply:
No. I added it to release.md. Also, I deleted this comment because it was wrong.

Reply:
But now yes? The only thing that should be built in release mode is the binary we pass to datagen. Everything else can run in debug mode.

Reply:
In most cases we're building only 1 crate. The only one where we run 2 builds is when installing icu4x-datagen, which has a separate discussion thread. I think this thread is therefore not actionable.